tests/pytorch_tests/model_tests/feature_models/reuse_layer_net_test.py

```diff
@@ -12,56 +12,164 @@
+        raise NotImplementedError("Subclasses must implement forward method")
+
+class ReuseLayerNet(BaseReuseLayerNet):
+    def __init__(self):
+        super().__init__()
+        self.conv1 = nn.Conv2d(3, 3, kernel_size=1, stride=1)
+        self.conv2 = nn.Conv2d(3, 3, kernel_size=1, stride=1)
+
+    def forward(self, x):
         x = self.conv1(x)
-        x = self.identity(x)
         x = self.conv2(x)
-        x = self.identity(x)
+        x = self.conv1(x)
         x = self.conv2(x)
-        x = self.identity(x)
         x = self.conv2(x)
-        x = self.identity(x)
-        y = self.conv2(y)
-        y = self.identity(y)
-        y = self.conv2(y)
-        y = self.identity(y)
-        y = self.conv1(y)
-        y = self.identity(y)
-        y = self.conv1(y)
-        y = self.identity(y)
-        y = self.conv1(y)
-        return x - y, y - x
-
-
-class ReuseLayerNetTest(BasePytorchTest):
-    """
-    This test checks:
-    The reuse of a layer in a model.
-    """
+        return x
+
+class ReuseFunctionalLayerNet(BaseReuseLayerNet):
+    def __init__(self):
+        super().__init__()
+        self.conv1_weight = nn.Parameter(torch.randn(3, 3, 2, 2))
+        self.conv1_bias = nn.Parameter(torch.randn(3))
+        self.conv2_weight = nn.Parameter(torch.randn(3, 3, 2, 2))
+        self.conv2_bias = nn.Parameter(torch.randn(3))
+
+    def forward(self, x):
+        x = F.conv2d(x, self.conv1_weight, self.conv1_bias, stride=1, groups=1)
+        x = F.conv2d(x, self.conv1_weight, self.conv1_bias, stride=1, groups=1)
+        x = F.conv2d(x, self.conv2_weight, self.conv2_bias, stride=1, groups=1)
+        x = F.conv2d(x, self.conv2_weight, self.conv2_bias, stride=1, groups=1)
+        return x
+
+class CombinedReuseNet(BaseReuseLayerNet):
+    def __init__(self):
+        super().__init__()
+        self.conv1 = nn.Conv2d(3, 3, kernel_size=3, stride=1, padding=1)
+        self.func_conv_weight = nn.Parameter(torch.randn(3, 3, 3, 3))
+        self.func_conv_bias = nn.Parameter(torch.randn(3))
+
+    def forward(self, x):
+        x = self.conv1(x)
+        x = self.conv1(x)
+        x = self.conv1(x)
+        x = F.conv2d(x, self.func_conv_weight, self.func_conv_bias, stride=1, padding=1)
+        x = F.conv2d(x, self.func_conv_weight, self.func_conv_bias, stride=1, padding=1)
+        return x
+
+class ReuseLayerTestBase(BasePytorchTest):
     def __init__(self, unit_test):
         super().__init__(unit_test)
 
     def create_inputs_shape(self):
-        return [[self.val_batch_size, 3, 32, 32], [self.val_batch_size, 3, 32, 32]]
+        return [[self.val_batch_size, 3, 32, 32]]
 
     def create_feature_network(self, input_shape):
-        return ReuseLayerNet()
\ No newline at end of file
+        return self.get_model()
+
+    def get_model(self):
+        raise NotImplementedError("Subclasses must implement get_model method")
+
+    def compare(self, quantized_models, float_model, input_x=None, quantization_info=None):
```

Comment:
Consider implementing explicit compare in the concrete tests. IMO in tests simplicity and clarity is more important than a perfect OOP.